Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -374,18 +410,65 @@ open class NavigationMapView: UIView {
         
         mostRecentUserCourseViewLocation = location
         
-        // While animating to overview mode, don't animate the puck.
-        let duration: TimeInterval = animated && navigationCamera.state != .transitionToOverview ? 1 : 0
-        UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear]) { [weak self] in
-            guard let screenCoordinate = self?.mapView.screenCoordinate(for: location.coordinate) else { return }
-            self?.userCourseView.center = CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
+        switch puckType {
+        case .courseView(configuration: _):
+            // While animating to overview mode, don't animate the puck.
+            let duration: TimeInterval = animated && navigationCamera.state != .transitionToOverview ? 1 : 0
+            UIView.animate(withDuration: duration, delay: 0, options: [.curveLinear]) { [weak self] in
+                guard let screenCoordinate = self?.mapView.screenCoordinate(for: location.coordinate) else { return }
+                self?.userCourseView.center = CGPoint(x: screenCoordinate.x, y: screenCoordinate.y)
+            }
+            
+            let cameraOptions = CameraOptions(cameraState: mapView.cameraState)
+            userCourseView.update(location: location,
+                                  pitch: cameraOptions.pitch!,
+                                  direction: cameraOptions.bearing!,
+                                  animated: animated,
+                                  navigationCameraState: navigationCamera.state)
+        case .puck2D(configuration: _):
+            if simulation {
+                if !(mapView.location.locationProvider is PassiveLocationManager) {
+                    mapView.location.locationProvider.stopUpdatingLocation()
```

Comment:
I don’t quite understand why we’d only shut off the map view’s built-in location provider when using SimulatedLocationManager (since that’s the only case when `simulation` is true). Is it because PassiveLocationManager and ReplayLocationManager don’t conform to LocationProvider? If so, perhaps MapboxNavigation could extend these classes to conform to LocationProvider to eliminate this and other special cases.

Reply:
This also calls into question whether we need separate PassiveLocationDataSource and PassiveLocationManager classes. Maybe there should be just PassiveLocationManager, with the LocationProvider conformance implemented in an extension in MapboxNavigation.

Reply:
Because during the turn-by-turn navigation simulation, the default map view’s built-in location provider is the AppleLocationProvider.  If we increase the simulation speed, the location from the AppleLocationProvider would be different from the routeProgress of the simulation. And the heading from the  built-in location provider would be `nil`. If we don't shut off the map view’s built-in location provider, the puck would receive both updates from the AppleLocationProvider as a LocationConsumer,  and the SimulatedLocationManager through route progress, which cause it shifting in location and also twinkling in rotation.

Reply:
Is there a situation where NavigationMapView should continue to use AppleLocationProvider? If we’re always trying to prevent AppleLocationProvider from being active, then we could shut it off more directly:

```swift
// No need for simulatesLocation flag
if let appleLocationProvider = mapView.location.locationProvider as? AppleLocationProvider {
    appleLocationProvider.stopUpdatingLocation()
}
```

Reply:
There's a case that the `NavigationMapView` continue using `AppleLocationProvider`, that's during the turn-by-turn navigation with active guidance, not under simulation mode. And in this case, the `navigationMapView.updateUserCourseView` will still be called, to update the `navigationMapView.mostRecentUserCourseViewLocation`, which will be used for future vanishing route line. And the 2D puck or the 3D puck would not be skipped from the routeController updates, but only receiving the true location update from the `AppleLocationProvider`.